portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,299 @@
+ *    A single or an array of contact data, which could be a string or an object with text and a link.
+ * 
+ * @returns {!DocumentFragment} The populated entry.
+ */
+function loadContactEntry(entry, title, data) {
+  entry.querySelector('.title').append(title);
+
+  if (Array.isArray(data)) {
+    valueDivs = data.map(createContactEntryContent);
+    entry.querySelector('.content').append(...valueDivs);
+  } else {
+    const div = createContactEntryContent(data);
+    entry.querySelector('.content').append(div);
+  }
+
+  return entry;
+}
+
+/**
+ * This function creates either a div or an anchor element,
+ * based on the provided data.
+ * 
+ * @param {!(string | {text: string, link: string})} data 
+ *    Contact information to be used in the created element
+ * 
+ * @returns {!HTMLElement} The element containing contact information
+ */
+function createContactEntryContent(data) {
+  const node = document.createElement('div');;
+
+  if (typeof data === 'object') {
+    const { text, link } = data;
+
+    const a = document.createElement('a');
+    a.href = link;
+    a.append(text);
+
+    node.append(a);
+  } else {
+    node.append(data);
+  }
+
+  return node;
+}
+
+/**
+ * Processes and appends the given experience section information on the website body.
+ * 
+ * @param {!object} sectionData Dictionary containing the experience data to be shown. 
+ *    Its keys contain the name of each experience section, and its values are
+ *    an array of objects describing individual experiences.
+ */
+function loadSectionData(sectionData) {
+  const sectionContainer = document.querySelector('main');
+  const sections = createSections(sectionData);
+
+  sectionContainer.append(...sections);
+}
+
+/**
+ * This function converts each element in the given section data to 
+ * an html node based on the "section" template.
+ * 
+ * @param {!object} sectionData Dictionary containing the experience data to be shown. 
+ *    Its keys contain the name of each experience section, and its values are
+ *    an array of objects describing individual experiences.
+ * 
+ * @returns {!Array} The array of "section" html nodes.
+ */
+function createSections(sectionData) {
+  const sectionTemplate = document.querySelector('#section');
+  const sections = Object.entries(sectionData).map(
+    ([title, entryData]) => loadSection(sectionTemplate.content.cloneNode(true), title, entryData)
+  );
+
+  return sections;
+}
+
+/**
+ * This function populates an empty "section" html node given the section data. 
+ * 
+ * @param {!DocumentFragment} section The empty section html node.
+ * @param {!string} title The title of the current section.
+ * @param {!Array<object>} entryData An array of objects describing individual experiences
+ *    in each section.
+ * 
+ * @returns {!DocumentFragment} The populated section.
+ */
+function loadSection(section, title, entryData) {
```

Comment:
maybe `buildSection()`?

Reply:
Response in last comment about this

Reply:
Function names were changed in the latest commits. How about now?